letsencrypt-auto-source/pieces/bootstrappers/rpm_python3_legacy.sh

```diff
@@ -5,12 +5,11 @@ BOOTSTRAP_RPM_PYTHON3_LEGACY_VERSION=1
 EnablePython36SCL() {
   set +e
   # Do nothing if Python 3.6 is already available
-  ! "$EXISTS" python3.6 > /dev/null 2> /dev/null || return
+  "$EXISTS" python3.6 > /dev/null 2> /dev/null && set -e && return
   # Do nothing if SCL Python 3.6 is not installed
-  scl --list 2>/dev/null | grep -q rh-python36 || return
-  # Install SCL rh-python36
-  . scl_source enable rh-python36
-  set -e
+  ! scl --list 2>/dev/null | grep -q rh-python36 && set -e && return
+  # Enable SCL rh-python36
+  . scl_source enable rh-python36 && set -e
```

Comment:
Why not just:
```
"$EXISTS" python3.6 > /dev/null 2> /dev/null && return
scl --list 2>/dev/null | grep -q rh-python36 || return
set +e
. scl_source enable rh-python36
set -e
```
Although I'd personally prefer the more conventional (at least when not programming in bash):
```
if "$EXISTS" python3.6 > /dev/null 2> /dev/null; then
    return
fi
if ! scl --list 2>/dev/null | grep -q rh-python36; then
    return
fi
set +e
. scl_source enable rh-python36
set -e
```

Reply:
Ok for the second proposal. In fact the `set +e` at the beginning of the function was here to avoid letsencrypt-auto script to break when one of the assertion fails (which is expected). But using a classic `if` branch seems to absorb the error, and do not break the script even if we are still under `set -e` mode.

Reply:
I personally like the more verbose syntax with `if` statements anyway, but you're right that my first example doesn't work.

Looking at it again taught me something that makes we want to suggest one final change though. I initially thought my first example could be salvaged by changing the logic of the first statement to use `||` instead of `&&` but that doesn't work either. The reason for this is when `return` isn't given a value, it returns the value of the previous command according to https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_24. This means that in the case when the first command fails, `return` will use an error status causing the command to fail.

In the new situation with `if` statements, what is the value used by `return`? I personally have no idea and quickly testing it it seems fine, but to be safe, I think we should use `return 0`.

Reply:
Well, I forgot that functions in bash are totally counter-intuitive compared to most programming languages, as they cannot return arbitrary values, only exit codes. In that regard it makes sense that a return with parameter needs to return an exit code, and this can only be the exit code of the latest instruction executed ...

Let's be deterministic. I changed to `return 0`